app/views/contacts/index.html.erb

```diff
@@ -2,35 +2,11 @@
 
 <% if user_signed_in? %>
 
-  <p><%= button_to 'View my requests', user_contact_requests_path(current_user), method: :get, :class => "btn btn-outline-primary"  %></p>
-
-  <% @contacts.each do |contact| %>
-    <a class="card" href=<% user_contacts_path(contact) %>>
-      <div class="card-body">
-        <div class="container">
-          <div class="row">
-            <div class="col-xs">
-              <img src="//lh3.googleusercontent.com/-6V8xOA6M7BA/AAAAAAAAAAI/AAAAAAAAAAA/rzlHcD0KYwo/photo.jpg?sz=120" class="rounded-circle" style="width: 50px">
-            </div>
-            <div class="col-10">
-              <h5 class="card-title"><%= contact.name %></h5>
-              <p class="card-link"><i><%= contact.email %></i></p>
-            </div>
-          </div>
-          <% if contact.social_accounts.any? %>
-            <div class="row">
-              <% contact.social_accounts.each do |social_account| %>
-                <div class="col">
-                  <a href=<%= generate_link(social_account) %>>
-                    <i class="fa <%= generate_icon(social_account) %> fa-4x" aria-hidden="true"></i>
-                  </a>
-                </div>
-              <% end %>
-            </div>
-          <% end %>
-        </div>
-      </div>
-    </a>
-    <br>
-  <%end %>
+  <p><%= button_to 'View my requests', user_contact_requests_path(current_user), method: :get, :class => "btn btn-outline-primary" %></p>
+  <div>
+    <% @contacts.each do |contact| %>
+      <%= render partial: 'users/card', locals: { user: contact, mode: :overview } %>
```

Comment:
was genau macht das partial?

Reply:
It says where to find the rest of the html code to render those user cards. So you can put parts of your html in another file and it gets more managable.
See also [the documentation](https://guides.rubyonrails.org/layouts_and_rendering.html#using-partials)